File: struclin/linkedlist/linkedlist.py
class Node:
    def __init__(self, value = None):
        self.value = value
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None


    def __iter__(self):
        node = self.head

        while node is not None:
            yield node
            node = node.next


    def insertAtBegining(self, node):
        node.next = self.head
        self.head = node


    def isEmpty(self):
        if not self.head:
            return 1
        return 0


    def removeNode(self, node_to_remove):
        if self.head == node_to_remove:
            self.head = self.head.next
            return

        previous_node = self.head

        for node in self:
            if node == node_to_remove:
                previous_node.next = node.next
                return
            previous_node = node


myLinkedList = LinkedList()

firstNode = Node(1998)
secondNode = Node(2006)
badNode = Node(2020)
thirdNode = Node(2018)

secondNode.next = badNode
badNode.next = thirdNode

myLinkedList.head = secondNode

# insert operation
myLinkedList.insertAtBegining(firstNode)

# remove operation
myLinkedList.removeNode(badNode)

# front operation
print(myLinkedList.head.value)

# isEmpty operation
print(myLinkedList.isEmpty())

# show linked list
for node in myLinkedList:
    print(node.value)

File: struclin/linkedlist/linkedlist.js
class Node {
    constructor (value = null) {
        this.value = value
        this.next = null
    }
}

class LinkedList {
    constructor () {
        this.head = null
    }

    * nodes() {
        let node = this.head

        while (node != null){
            yield node
            node = node.next
        }
    }

    insertAtBegining(node) {
        node.next = this.head
        this.head = node
    }

    isEmpty() {
        if(!this.head) return 1

        return 0
    }

    removeNode(nodeToRemove) {
        if (this.head === nodeToRemove) {
            this.head = this.head.next
            return
        }

        let previous_node = this.head

        for(let node of this.nodes()) {
            if (node === nodeToRemove) {
                previous_node.next = node.next
                return
            }

            previous_node = node
        }
    }
}

const myLinkedList = new LinkedList()

firstNode = new Node(1998)
secondNode = new Node(2006)
badNode = new Node(2020)
thirdNode = new Node(2018)

secondNode.next = badNode
badNode.next = thirdNode

myLinkedList.head = secondNode

// insert operation
myLinkedList.insertAtBegining(firstNode)

// remove operation
myLinkedList.removeNode(badNode)

// front operation
console.log(myLinkedList.head.value)

// isEmpty operation
console.log(myLinkedList.isEmpty())

// show linked list
for(let node of myLinkedList.nodes()) {
    console.log(node.value)
}